sstables/sstable_set.cc

```diff
@@ -31,16 +31,26 @@ extern logging::logger sstlog;
 
 bool
 sstable_first_key_less_comparator::operator()(const shared_sstable& s1, const shared_sstable& s2) const {
-    return s1->compare_by_first_key(*s2) < 0;
+    auto r = s1->compare_by_first_key(*s2);
+    if (r == 0) {
+        position_in_partition::less_compare less_cmp(*s1->get_schema());
+        return less_cmp(s1->first_key_clustering_range().start(), s2->first_key_clustering_range().start());
```

Comment:
Shouldn't we compare `s1->last_key_clustering_range().end()` with `s2->first_key_clustering_range().start()`?